### PR TITLE
Side Panel Component

### LINKED ISSUE #3499: Side Panel Component
## 🗒 General

Would love some help building out this side panel component for Storybook! The side panel component can be triggered by a clickable component or a row of table to display additional information, with or without a lightbox in the background.

Side Panel

### Side Panel with Lightbox
- User can click the trigger to open the side panel, which slides in from the right
- There is a lightbox in the background of the side panel and block the page content
- User can click the close button or the lightbox to close it, and the panel will disappear by sliding over to the right

Has Lightbox_1
Has Lightbox_2

### Side Panel without Lightbox
- User can click the trigger to open the side panel, which slides in from the right
- There is no lightbox in the background of the side panel and user can scroll and interact with page content while side panel is open
- User can click on another button (‘trigger’ area) in the main page to swap the content of the side panel without closing it (e.g. click various rows of a table to view different details in side panel)
- User can click the close button or the “non trigger” area outside the side panel to close it, and the panel will disappear by sliding over to the right

No Lightbox_1
No Lightbox_2
No Lightbox_3

## 📐 Specs
You can access specs by duplicating our [Figma UI Kit](https://www.figma.com/file/nUaAw56hTgC0RIOYkuLSrB/Backstage-Design-System?node-id=1145%3A0). Click on the 'Side Panel' page on the left hand side.

Spec Img

### PR DESCRIPTION
**Pull request by ayshiff**
## Hey, I just made a Pull Request!

Closes #3499

This PR adds a new storybook entry for the [Drawer](https://material-ui.com/components/drawers/) component.
It documents the way to use it to match the desired design.
I tried to put the minimum logic and style to keep the code clear, with the requested UI and behaviour.

## Side Panel

drawer

## Behaviour

drawer

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [X] Screenshots attached (for UI changes)

### CODE REVIEW

File: packages/core/src/components/Drawer/Drawer.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+export default {
+  title: 'Layout/Drawer',
+  component: Drawer,
+};
+
+const useDrawerStyles = makeStyles({
+  paper: {
+    width: '50%',
+    justifyContent: 'space-between',
+    padding: '20px',
```
Comment (Rugvip) [inline packages/core/src/components/Drawer/Drawer.stories.tsx:36]: Use `theme.spacing` instead

File: packages/core/src/components/Drawer/Drawer.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+  },
+  icon: {
+    fontSize: 20,
+  },
+  content: {
+    height: '80%',
+    backgroundColor: '#EEEEEE',
+  },
+  secondaryAction: {
+    marginLeft: '20px',
```
Comment (Rugvip) [inline packages/core/src/components/Drawer/Drawer.stories.tsx:54]: `theme.spacing` here as well

**Review by Rugvip — COMMENTED** [review]
Thanks! a minor nit but otherwise good to go!

We've been discussing whether this should be its own component or not, but either way it makes sense to get this into storybook to establish the pattern and then perhaps provide something out of the box later.

**Review by Rugvip — APPROVED** [review]
Thanks! 😁  :shipit:

**Comment by stefanalund** [comment]
Nice work on this!

**Pull request merged**